### DISCUSSION TITLE
Week 1 Jun 17-21

### DISCUSSION THREAD

**Discussion by eeholmes** [SAFS-Varanasi-Internship/Summer-2024#10] [General]
* Try Jiarui's tutorials (Y & S)
* Research how to make a simple multiple variable ML model (any method) -- Y & S
* Get some zooplankton layers or decide on different predictor variable --- Eli
* Move Zarr file into cloud storage --- Eli
* Get GPU set up --- Eli

**Comment by eeholmes** [comment]
@NaNa7Miiii Questions for your tutorial

1) Is the tutorial on dhub shared directory? We did not find it.
2) The url to the zarr files is also not in the shared.
<img width="730" alt="image" src="https://github.com/SAFS-Varanasi-Internship/Summer-2024/assets/2545978/4c5c9995-d596-49ba-a614-d532f06dbda9">

Questions for your original models for predicting upwelling.

Can you tell us which notebooks are for your upwelling predictions?

**Comment by NaNa7Miiii** [comment]
Hi Dr.Eli
Questions for tutorial:
1. I have sent the tutorial files via email before, I have also attached
them in this email (this contains a land mask file)
2. The zarr files were created by Minh last summer and are located in the
path '/home/jovyan/shared/data/INDIAN_OCEAN_025GRID_DAILY.zarr'
(I downloaded that zarr file so I have a different path)

Question for original model:
The tutorial.ipynb file is for SST prediction, and for upwelling
predictions using CNN I was using a file called 2D_helper:
-> shared -> jiarui-notebooks -> 2D_helper.ipynb
Best,
Jiarui

**Reply by eeholmes** [reply]
Thanks!

**Comment by eeholmes** [comment]
Thurday:
Working on possible predictor variables.
UC Santa Cruz West Coast ROMS:
https://oceanmodeling.ucsc.edu/ccsra_nem_2017a/#txtOutput

West Coast ROMS: https://data.caloos.org/#module-metadata/0ac0326a-a2c6-11e2-9dc2-00219bfe5678
Modelled zooplankton: https://oceanmodeling.ucsc.edu/thredds/dodsC/CCSRA_NEM_2017a_agg_catalog_zlevs/fmrc/CCSRA_NEM_2017a_Historical_Reanalysis_Phys-Bio_ROMS_NEMURO_Aggregation_best.ncd.html

Bering Sea ACLIM
https://data.pmel.noaa.gov/aclim/thredds/catalog/catalog.html
https://beringnpz.github.io/roms-bering-sea/B10K-dataset-docs/

**Comment by eeholmes** [comment]
@yifeihang @ShridharS19 I will be in at 3pm for ERDDAP training and we will continue project discussion afterwards.

**Comment by ShridharS19** [comment]
Week 1 log (06/20/2024) :
- Got the ConvLSTM Tutorial running
- Read through and ran the EfficeintNet Model tutorial for Upwelling

**Comment by yifeihang** [comment]
06/20/24

- Got the ConvLSTM tutorial running on JupyterHub (but each epoch is taking about 16 min with CPU so I didn't run the whole thing)
- Went through and ran the ml_tutorials 3 and 4.
- Working on getting CUDA set up on my local machine.

Question about tutorial 04 (Path: shared/jiarui-notebooks/ml_tutorials/04_EfficientNet.ipynb):
- Under data selection, the tutorial says the output for sst_data.shape is supposed to be (1218, 101, 41, 1) while I got (1218, 161, 181, 1). What would be causing the changes in long and lat dimensions?

**Comment by eeholmes** [comment]
More ML models to explore:

* https://geosmart-2023.hackweek.io/tutorials/snow_cover_mapping/SCA_Mapping.html
* https://geosmart-2023.hackweek.io/tutorials/snow-extrapolation/Introduction.html
* https://geosmart-2023.hackweek.io/tutorials/snow-extrapolation/Introduction.html#

Info on the Indian Ocean Zarr file:

https://hackweek-itcoocean.github.io/2023-Hackbook/python-notebooks/02_Using_Data.html

**Comment by yifeihang** [comment]
Fri. 06/21/24

- Continued on CUDA set up: tensorflow can now detect my GPU, but I'm still working on using it in Jupyter notebook
- Went through some of Minh's tutorials on how to use the zarr file he built, interacted with the data

Questions:
- Which of the variables is for bathymetry?
- I've been taking fragmented notes while installing CUDA. Should I write up a tutorial kind of thing for the installation process?

**Comment by eeholmes** [comment]
Yes please write up a tutorial re how you set up your computer. This will
be very helpful for future interns!

Re bathymetry. Minh called it 'topo'

**Reply by yifeihang** [reply]
Got it, thank you
